ID: 57587
Title: panic in TestGetMaxRunningTasksLimit 
Description:
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
see https://do.pingcap.net/jenkins/blue/rest/organizations/jenkins/pipelines/pingcap/pipelines/tidb/pipelines/ghpr_unit_test/runs/22863/nodes/63/steps/69/log/?start=0
```
=== RUN   TestGetMaxRunningTasksLimit
[2024/11/21 07:41:31.629 +00:00] [ERROR] [region_cache.go:685] ["cluster id is not set properly"] [stack="github.com/tikv/client-go/v2/internal/locate.NewRegionCache\n\texternal/com_github_tikv_client_go_v2/internal/locate/region_cache.go:685\ngithub.com/tikv/client-go/v2/tikv.NewRegionCache\n\texternal/com_github_tikv_client_go_v2/tikv/region.go:211\ngithub.com/pingcap/tidb/pkg/ttl/ttlworker.TestGetMaxRunningTasksLimit\n\tpkg/ttl/ttlworker/task_manager_test.go:228\ntesting.tRunner\n\tGOROOT/src/testing/testing.go:1690"]
--- PASS: TestGetMaxRunningTasksLimit (0.00s)
PASS
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x3862f03]

goroutine 122 [running]:
github.com/tikv/client-go/v2/internal/locate.(*storeCacheImpl).fetchAllStores(0xc001edeb00, {0xa544708, 0xc000e0e280})
	external/com_github_tikv_client_go_v2/internal/locate/store_cache.go:121 +0x43
github.com/tikv/client-go/v2/internal/locate.refreshFullStoreList({0xa544708, 0xc000e0e280}, {0xa57fee0, 0xc001edeb00})
	external/com_github_tikv_client_go_v2/internal/locate/region_cache.go:740 +0x70
github.com/tikv/client-go/v2/internal/locate.NewRegionCache.func3({0xa544708, 0xc000e0e280}, {0x0?, 0x0?, 0x0?})
	external/com_github_tikv_client_go_v2/internal/locate/region_cache.go:731 +0x5d
github.com/tikv/client-go/v2/internal/locate.(*bgRunner).schedule.func1()
	external/com_github_tikv_client_go_v2/internal/locate/region_cache.go:574 +0x25a
created by github.com/tikv/client-go/v2/internal/locate.(*bgRunner).schedule in goroutine 117
	external/com_github_tikv_client_go_v2/internal/locate/region_cache.go:563 +0x148
--
Coverage runner: Not collecting coverage for failed test.
The following commands failed with status 2
/home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/sandbox/linux-sandbox/21343/execroot/__main__/bazel-out/k8-fastbuild/bin/pkg/ttl/ttlworker/ttlworker_test_/ttlworker_test.runfiles/__main__/pkg/ttl/ttlworker/ttlworker_test_/ttlworker_test
```

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)
found in this pr https://github.com/pingcap/tidb/pull/57557, but seems not directly related to it
<!-- Paste the output of SELECT tidb_version() -->

